chaintesting/action_test_helpers.go

```diff
@@ -0,0 +1,81 @@
+// Copyright (C) 2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package chaintesting
+
+import (
+	"context"
+	"testing"
+
+	"github.com/ava-labs/avalanchego/database"
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/stretchr/testify/require"
+
+	"github.com/ava-labs/hypersdk/chain"
+	"github.com/ava-labs/hypersdk/codec"
+	"github.com/ava-labs/hypersdk/state"
+)
+
+var _ state.Mutable = (*InMemoryStore)(nil)
+
+// InMemoryStore is a storage that acts as a wrapper around a map and implements state.Mutable.
+type InMemoryStore struct {
+	Storage map[string][]byte
+}
+
+func NewInMemoryStore() *InMemoryStore {
+	return &InMemoryStore{
+		Storage: make(map[string][]byte),
+	}
+}
+
+func (s *InMemoryStore) GetValue(_ context.Context, key []byte) ([]byte, error) {
+	val, ok := s.Storage[string(key)]
+	if !ok {
+		return nil, database.ErrNotFound
+	}
+	return val, nil
+}
+
+func (s *InMemoryStore) Insert(_ context.Context, key []byte, value []byte) error {
+	s.Storage[string(key)] = value
+	return nil
+}
+
+func (s *InMemoryStore) Remove(_ context.Context, key []byte) error {
+	delete(s.Storage, string(key))
+	return nil
+}
+
+// ActionTest is a single parameterized test. It calls Execute on the action with the passed parameters
+// and checks that the assertions passes.
+type ActionTest struct {
+	Action chain.Action
+
+	Rules     chain.Rules
+	State     state.Mutable
+	Timestamp int64
+	Actor     codec.Address
+	ActionID  ids.ID
+
+	ExpectedOutputs [][]byte
+	ExpectedErr     error
+}
+
+type ActionTestSuite struct {
+	Tests map[string]ActionTest
+}
```

Comment:
Could we move this to `chain/` so that it's in the same package as the interface it's used to test?

Reply:
@ARR4N told me to move it from `chain/` to `chaintesting/` in https://github.com/ava-labs/hypersdk/pull/1139#discussion_r1679256896 since these are test-only constructs, to avoid them leaking to production.

Reply:
Ah, will defer so this can be merged and start a separate conversation on what convention we should go with